pkg/kubelet/kubelet_node_status.go

```diff
@@ -551,20 +552,17 @@ func (kl *Kubelet) updateNodeStatus(ctx context.Context) error {
 
 // tryUpdateNodeStatus tries to update node status to master if there is any
 // change or enough time passed from the last sync.
-func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error {
-	// In large clusters, GET and PUT operations on Node objects coming
-	// from here are the majority of load on apiserver and etcd.
-	// To reduce the load on etcd, we are serving GET operations from
-	// apiserver cache (the data might be slightly delayed but it doesn't
-	// seem to cause more conflict - the delays are pretty small).
-	// If it result in a conflict, all retries are served directly from etcd.
```

Comment:
This was an educative comment on the apiserver load and usage of cached object. Could we something similar for the new code, e.g. in the method "docstring" comment or somewhere. WDYT?

Reply:
Yeah, I think the comment is still relevant after the change. Let me rephrase a bit.

Reply:
@wojtek-t could you help confirm whether the approach -- only get from etcd directly when `apierrors.IsConflict()`-- is desired? Thanks!

Reply:
I wasn't aware of this PR...

It is and in fact already happened in: https://github.com/kubernetes/kubernetes/pull/123977, but it's desired on the first attempt. If we failed to update it (which BTW in practice is rare), we should rather get from apiserver in such case.
I think that doing that only for IsConflict() will not bring any non-negligible impact and introduces another complexity, so I wouldn't do that.

And if we exclude that, I think this PR is effectively no-op [on top of the my PR merged few months ago]

Reply:
Got it. Agree that the main benefit already happened with the previous PR. Given the unclear impact of this PR, @linxiulei maybe we can close this PR.

Reply:
@yujuhong @wojtek-t Let me clarify the value of this PR. From my observation, especially in cases when a cluster is being upscaled rapidly, updates are likely to be failed with 429s (i.e. apiserver couldn't serve that much requests), then in current design the kubelet would get the Node object from etcd, which compounds the situation.

So the value of the PR is to avoid such cases and make whole system is recoverable rather than kubelets keeping inflicting overhead to apiserver.

Reply:
are those 429 not sending Retry-after headers?
https://github.com/kubernetes/kubernetes/blob/5bf1e95541d90e37f6c6637b5b45d8783e7907aa/staging/src/k8s.io/apiserver/pkg/server/filters/with_retry_after.go#L55-L65

Reply:
I don't think so, because regardless of how the node is retrieved from, the update may be failed with 429s, right? This failure is not because of staleness but apiserver's being overwhelmed. Getting node objects from etcd doesn't really help but make the situation worse.

Reply:
429s are automatically retries (IIRC 10 times by default) as Antonio noted. 
I would like to avoid over-optimizing as it always introduces additional complexity - let's see how much the already merged improvement helps and make a more data-driven decision after that.

Reply:
Well, then I think this PR can wait until we see the need. Thanks all!